src/site/shared/utils/SnapUtils.ts

```diff
@@ -0,0 +1,40 @@
+import {Component} from "core/public";
+import {V, Vector} from "Vector";
+
+
+const WIRE_SNAP_THRESHOLD = 0.2;
+
+/**
+ * Utility used for calculating positions of ports and wires when a component
+ * gets moved.
```

Comment:
Can you describe the behavior more specifically.  i.e. "Adjusts a proposed new position for a component to snap it into positions that result in axis-aligned wire connections."

Reply:
Yeah.. this is legacy code and this comment was probably written as RCOS fodder